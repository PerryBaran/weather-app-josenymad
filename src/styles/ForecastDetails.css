.forecast-details {
    text-align: center;
    background-color: cadetblue;
    width: 30vw;
    margin: auto;
    border-radius: 15vw;
}

.forecast-details__container {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    height: 30vh;
}