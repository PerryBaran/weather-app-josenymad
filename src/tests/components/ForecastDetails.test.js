import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 30,
    wind: {
      speed: 60,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Mon Apr 30 2018")).toBeInTheDocument();
    expect(getByText("ðŸŒ¡ Top temperature: 22Â°C")).toBeInTheDocument();
    expect(getByText("ðŸŒ¡ Minimum temperature: 12Â°C")).toBeInTheDocument();
    expect(getByText("ðŸ’§ Humidity: 30%")).toBeInTheDocument();
    expect(getByText("ðŸ’¨ Wind speed: 60mph")).toBeInTheDocument();
    expect(getByText("ðŸ§­ Wind direction: N")).toBeInTheDocument();
  });
});
