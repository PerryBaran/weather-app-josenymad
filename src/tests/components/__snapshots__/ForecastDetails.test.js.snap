// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h3
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </h3>
    <div
      class="forecast-details__container"
    >
      <div
        class="forecast-details__max-temp"
      >
        ðŸŒ¡ Top temperature: 22Â°C
      </div>
      <div
        class="forecast-details__min-temp"
      >
        ðŸŒ¡ Minimum temperature: 12Â°C
      </div>
      <div
        class="forecast-details__humidity"
      >
        ðŸ’§ Humidity: 30%
      </div>
      <div
        class="forecast-details__wind-speed"
      >
        ðŸ’¨ Wind speed: 60mph
      </div>
      <div
        class="forecast-details__wind-direction"
      >
        ðŸ§­ Wind direction: N
      </div>
    </div>
  </div>
</DocumentFragment>
`;
